### DISCUSSION TITLE
How to localize DateRange filter

### DISCUSSION THREAD

**Discussion by qwasyx0** [rappasoft/laravel-livewire-tables#1661] [General]
The DateRange component uses flatpickr and in vendor/rappasoft/laravel-livewire-tables/resources/js/partials/filter-date-range.min.js I see that locale is set to "en". So far I have tried to set the language as they say in [docs](https://flatpickr.js.org/localization/) and how they do it in [laravel-splade](https://github.com/protonemedia/laravel-splade/issues/449) but I can't seem to get it working.

Also, when date is selected, in filters, the months are shown also in english, even though my app locale is set to "cs". I have changed the format to "d/m/y" for now.

**Comment by lrljoe** [comment]
Firstly - thanks for sponsoring the package!

Can you give me an example of your DateRangeFilter, along with a screenshot of the behaviour you're seeing, so that I can try and replicate the issue on my end?

I do plan to expose all of the flatpickr methods for DateRangeFilter in a upcoming release, just not sure on timescales just yet, but can prioritise anything that isn't working for a sponsor.

Do feel free to give me a nudge on Discord if I'm not prompt about answering!

**Comment by qwasyx0** [comment]
Hello,

this is how I use DateRangeFilter:

```php
'DATUM' => DateRangeFilter::make('Datum vystavení', 'DATUM')
->config([
'dateFormat' => 'Y-m-d',
'placeholder' => 'Zadejte rozmezí',
])
->setFilterPillValues([0 => 'minDate', 1 => 'maxDate']),
```

in my app.php I have set locale to cs

```php
'locale' => 'cs',
```

I would expect the flatpickr to be translated into czech langauge, so
February=Únor and so on.

![screen](https://github.com/rappasoft/laravel-livewire-tables/assets/43924784/24b7e878-b384-40db-94f8-8fa424827534)


And the filter bubbles to be also localized to czech (it should read 01 Úno
2024 Až 29 Úno 2024

![screen](https://github.com/rappasoft/laravel-livewire-tables/assets/43924784/11f381c3-985a-49e3-8031-445ea45deb53)


Carbon is working fine as this line of code reads "února 26 2024, 1:52:04
dopoledne"
```php
Carbon\Carbon::now()->isoFormat('MMMM Do YYYY, h:mm:ss a')
```

**Comment by qwasyx0** [comment]
Hello @lrljoe any update on this?

**Reply by lrljoe** [reply]
Please reach out on Discord if you'd like an early access branch to test with, as I'm in the process of updating the docs for this.

**Comment by lrljoe** [comment]
So I'm actively looking at this at the moment.

The config option will be in the next version.

**Comment by lrljoe** [comment]
Merged into development, if you'd like to test

**Comment by lrljoe** [comment]
This is in the latest version
